test/powershell/Modules/Microsoft.PowerShell.Management/FunctionProvider.Tests.ps1

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "Basic Function Provider Tests" -Tags "CI" {
+    BeforeAll {
+        $existingFunction = "existingFunction"
+        $nonExistingFunction = "nonExistingFunction"
+        $text = "Hello World!"
+        $functionValue = { return $text }
+        $restoreLocation = Get-Location
+        Set-Location Function:
+    }
+
+    AfterAll {
+        Set-Location -Path $restoreLocation
+    }
+
+    Context "Validate Set-Item Cmdlet" {
+        BeforeEach {
+            Set-Item $existingFunction -Options "None" -Value $functionValue
+        }
+
+        AfterEach {
+            Remove-Item $existingFunction -ErrorAction SilentlyContinue -Force
+            Remove-Item $nonexistingFunction -ErrorAction SilentlyContinue -Force
+        }
+
+        It "Sets the new options in existing function" {
+            $newOptions = "ReadOnly, AllScope"
+            (Get-Item $existingFunction).Options | Should -Be "None"
+            Set-Item $existingFunction -Options $newOptions
+            (Get-Item $existingFunction).Options | Should -Be $newOptions
+        }
+
+        It "Sets the options and a value of type ScriptBlock for a new function" {
+            $options = "ReadOnly"
+            Set-Item $nonExistingFunction -Options $options -Value $functionValue
+            (Get-Item $nonExistingFunction).Options | Should -Be $options
```

Comment:
use a variable to store the value of `(Get-Item $nonExistingFunction)` and reuse instead of calling `Get-Item` twice.